src/main.cpp

```diff
@@ -2696,25 +2697,47 @@ static int64_t nTimeIndex = 0;
 static int64_t nTimeCallbacks = 0;
 static int64_t nTimeTotal = 0;
 
+/**
+ * Determine whether to do transaction checks when validating blocks.
+ * Returns `false` (allowing transaction checks to be skipped) only if all
+ * of the following are true:
+ *   - we're currently in initial block download
+ *   - the `-ibdskiptxverification` flag is set
+ *   - the block under inspection is an ancestor of the latest checkpoint.
+ */
+static bool IBDSkipTxVerification(const CChainParams& chainparams, const CBlockIndex* pindex) {
+    return IsInitialBlockDownload(chainparams)
+        && fIBDSkipTxVerification
+        && fCheckpointsEnabled
+        && Checkpoints::IsAncestorOfLastCheckpoint(chainparams.Checkpoints(), pindex);
+}
+
+/**
+ * - Validate the block with CheckBlock
+ * - Check the the CCoinsViewCache's best block is the parent of the block.
+ * -
+ */
```

Comment:
```suggestion
```
`ConnectBlock` is already documented in `main.h`; any changes should be made there.